cordova-lib/src/cordova/platform.js

```diff
@@ -86,37 +87,46 @@ function addHelper(cmd, hooksRunner, projectRoot, targets, opts) {
         
         var platformsToSave = []; 
 
-        // If statement to see if pkgJsonPath exists in the filesystem
-        
         return promiseutil.Q_chainmap(targets, function(target) {
             // For each platform, download it and call its helper script.
             var parts = target.split('@');
             var platform = parts[0];
-            var spec = parts[1];
-
+            spec = parts[1];
+            var pkgJson;
+            var pkgJsonPath = path.join(projectRoot, 'package.json');
+            if(fs.existsSync(pkgJsonPath)) {
+                delete require.cache[require.resolve(pkgJsonPath)]; 
+                pkgJson = require(pkgJsonPath);
+            }
             return Q.when().then(function() {
+                var pkgJsonSpec = 'cordova-'+platform;
                 if (!(platform in platforms)) {
                     spec = platform;
                     platform = null;
                 }
-
                 if(platform === 'amazon-fireos') {
                     events.emit('warn', 'amazon-fireos has been deprecated. Please use android instead.');
                 }
                 if(platform === 'wp8') {
                     events.emit('warn', 'wp8 has been deprecated. Please use windows instead.');
                 }
-                if (platform && !spec && cmd == 'add') {
+                if(spec && pkgJson) {
+                    pkgJson.dependencies[pkgJsonSpec] = spec;
+                }
+                // If there is no spec specified during add, use the one from pkg.json.
+                if(spec === undefined && pkgJson && pkgJson.dependencies && pkgJson.dependencies[pkgJsonSpec]) {
+                    spec = pkgJson.dependencies[pkgJsonSpec];
+                }
+                // If there is no spec in pkg.json, use config.xml.
+                if (platform && !spec && cmd == 'add' && !pkgJson || pkgJson && !pkgJson.dependencies || !pkgJson.dependencies[pkgJsonSpec]) {
```

Comment:
`(platform && !spec && cmd == 'add')` should suffice since spec would already have a value from the previous if statement if pkgJson.dependences[pkgJsonSpec] existed